ExtensionStore/lib/store.js

```diff
@@ -1247,41 +1248,68 @@ function ExtensionDownloader(extension) {
 
 /**
  * Downloads the files of the extension from the repository set in the object instance.
+ * @param {QToolButton} widget - UI icon to treat as a progressbar.
  * @returns [string[]]    an array of paths of the downloaded files location, as well as the destination folder at index 0 of the array.
  */
-ExtensionDownloader.prototype.downloadFiles = function () {
+ExtensionDownloader.prototype.downloadFiles = function (widget) {
   this.log.info("starting download of files from extension " + this.extension.name);
   var destFolder = this.destFolder;
-
-  var progress = new QProgressDialog();
-  progress.title = "Installing extension " + this.extension.name;
-  progress.setLabelText("Downloading files...");
-  progress.modal = true;
-  progress.show();
-
+  
   // get the files list (heavy operations)
   var destPaths = this.extension.localPaths.map(function (x) { return destFolder + x });
   var dlFiles = [this.destFolder];
   var files = this.extension.files;
-
-  progress.setRange(0, files.length);
+ 
   this.log.debug("downloading files : "+files.map(function(x){return x.path}).join("\n"))
 
-  for (var i = 0; i < files.length; i++) {
+  // Set parameters based on whether provided widget is the update button, or install extension button.
+  var widgetClass = widget.objectName === "installButton" ? "QToolButton" : "QPushButton";
```

Comment:
this for example is a massive separation of concerns red flag to me, all those buttons should be able to handle the stylesheet/representation part, not the ExtensionDownloader itself.
Maybe we should leave those buttons blank in the UI, then subclass QToolButton or QPushButton into a ProgressButton, then create those buttons at runtime ?